homeassistant/components/matter/entity.py

```diff
@@ -55,7 +56,122 @@ async def wrapper(self: MatterEntity, *args: P.args, **kwargs: P.kwargs) -> _R:
 
 
 @dataclass(frozen=True)
-class MatterEntityDescription(EntityDescription):
+class MatterEntityLabeling(EntityDescription):
+    """Data structure for Matter Tag and Label Information."""
+
+    # A label or tag can be used to modify an entity's name by appending the text
+    # or replacing the name. This class holds the information needed to do that.
+
+    # Where to place the Tag or Label text in the entity name.
+    # Can be "rename", "append", "ignore". Append by default.
+    label_placement: str = "append"
```

Comment:
I feel like it would make sense to define a `(Str)Enum` for these three label placement types.

Reply:
I could do that next week or as a small PR if main code is accepted before then.

Reply:
This has now been completed.